plugins/kosync.koplugin/main.lua

```diff
@@ -346,25 +432,68 @@ function KOSync:getProgress(manual)
             doc_digest,
             function(ok, body)
```

Comment:
This whole nested if branches is getting to deep and not readable anymore. Can we refactor it a little bit? Try convert chunk of code into helper functions or use the following pattern if applicable:

```lua
if not good then
   -- do error handling
   return
end

if not good2 then
  -- do more error handling
  return
end

-- everything is good, do stuff
```